convert2rhel/unit_tests/breadcrumbs_test.py

```diff
@@ -268,3 +268,23 @@ def test_set_target_os(pretend_os):
         "name": "CentOS Linux",
         "version": "7.9",
     } == breadcrumbs.breadcrumbs.target_os
+
+
+@pytest.mark.parametrize(("telemetry_disabled", "telemetry_called"), [(True, 0), (False, 1)])
+def test_disable_telemetry(telemetry_disabled, telemetry_called, monkeypatch):
+    if telemetry_disabled:
+        monkeypatch.setenv("CONVERT2RHEL_DISABLE_TELEMETRY", "1")
+
+    _set_ended = mock.Mock()
+    _save_migration_results = mock.Mock()
+    _save_rhsm_facts = mock.Mock()
+
+    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_set_ended", _set_ended)
+    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_save_migration_results", _save_migration_results)
+    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_save_rhsm_facts", _save_rhsm_facts)
+
+    breadcrumbs.breadcrumbs.finish_collection()
+
+    assert _set_ended.call_count == 1
+    assert _save_migration_results.call_count == 1
+    assert _save_rhsm_facts.call_count == telemetry_called
```

Comment:
I would say that mocking the `_set_ended` function is not necessarily at all. It will just return a timestamp string for us. 

```suggestion
    _save_migration_results = mock.Mock()
    _save_rhsm_facts = mock.Mock()

    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_save_migration_results", _save_migration_results)
    monkeypatch.setattr(breadcrumbs.breadcrumbs, "_save_rhsm_facts", _save_rhsm_facts)

    breadcrumbs.breadcrumbs.finish_collection()

    assert _save_migration_results.call_count == 1
    assert _save_rhsm_facts.call_count == telemetry_called
```